File: C#/Rotate.cs
public class Solution {
    public void Rotate(int[][] matrix) {
        
        List<int> OneDim = new List<int>();
        
        for (int row = 0; row < matrix.Length; row++)
        {
            for (int col = matrix[0].Length - 1; col > -1; col--)
            {   
                OneDim.Add(matrix[col][row]);
                
                // Debug
                Console.Write(matrix[col][row] + ", ");
            }
            
            Console.Write("\n");
        }
        Console.Write("\n");
        
        // Debug
        for (int i = 0; i < OneDim.Count; i++)
        {
            Console.Write(OneDim[i] + ", ");
        }
        Console.Write("\n\n");
        
        
        int index = 0;
        for (int row = 0; row < matrix.Length; row++)
        {
            for (int col = 0; col < matrix.Length; col++)
            {
                matrix[row][col] = OneDim[index];
                index++;
            }
        }
    }
}

File: Python/rotate.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        # Copy of the matrix we are rotating
        n = len(matrix)
        copy = []

        row = 0
        while (row < n):
            col = 0
            while (col < n):
                copy.append(matrix[row][col])
                col += 1
            row += 1  

        col = 0
        while (col < n):
            row = 0
            while (row < n):
                # print(str(row) + "," + str(col) + ": " + str(copy[(row * n) + col]))
                matrix[col][n - row - 1] = copy[(row * n) + col]
                row += 1
            col += 1